vtrace/util.py

```diff
@@ -76,3 +78,220 @@ def fireLocalNotifiers(self, event, trace):
+        elif psize == 8:
+            emu.setSegmentInfo(e_amd64.SEG_GS, seginfo, 0xffffffffffff)
+
+    return emu
+
+
+def vwFromTrace(trace, storagename='binary_workspace_from_vsnap.viv', filefmt=None, collapse=True, strict=True):
+    '''
+    Produce an envi emulator for this tracer object.
+
+    If filefmt is None, it will be auto-determined
+
+    If collapse, join adjacent maps
+    If strict, only join maps with the same permissions
+    '''
+    import vivisect
+    import envi.memory as e_memory
+    vw = vivisect.VivWorkspace()
+    arch = trace.getMeta('Architecture')
+    plat = trace.getMeta('Platform')
+    psize = trace.getPointerSize()
+
+    # determine file format (if not specified above)
+    if filefmt is None:
+        if 'win' in plat.lower():
+            filefmt = 'pe'
+            from vivisect.parsers.pe import archcalls
+        else:
+            filefmt = 'elf'
+            from vivisect.parsers.elf import archcalls
+
+    vw.setMeta("Architecture", arch)
+    vw.setMeta("Platform", plat)
+    vw.setMeta('Format', filefmt)
+    vw.setMeta('DefaultCall', archcalls.get(arch,'unknown'))
+    vw.setMeta('StorageName', storagename)
+    
+    if 'win' in plat.lower():
+        ossep = '\\'
+        exts = ('exe', 'dll')
+
+    else:
+        ossep = '/'
+        exts = ('so')
+
+    # could use {get,set}MemorySnap if trace inherited from MemoryObject
+    seen = []
+    maps = []
+    for va, size, perms, fname in trace.getMemoryMaps():
+        # strip off unwanted parts
+        trimfname = fname.split(ossep)[-1]
+        stripfname = trimfname
+        for ext in exts:
+            if trimfname.endswith('.' + ext):
+                stripfname = trimfname[:-(len(ext) + 1)]
+
+        # add map to the workspace
+        try:
+            # So linux maps in a PROT_NONE page for efficient library sharing, so we have to take that into account
+            if (not perms & e_memory.MM_READ):
+                continue
+            if plat == 'linux' and stripfname in ['[vvar]']:
+                continue
+            bytez = trace.readMemory(va, size)
+            maps.append((va, perms, stripfname, bytez))
+
+        except vtrace.PlatformException:
+            print('failed to map: 0x{:x} into emu'.format(va, size))
+            continue
+
+        # add file
+        if fname in seen: 
+            continue 
+        seen.append(fname) 
+        
+        # remove path from name
+        if trimfname in vw.getFiles(): 
+            continue
+
+        fakemd5 = 'adsfasdfasdfasdfasdf'    
+        # TODO? actually work through all the maps with the same name and add some validish MD5?
+        # but because we're yanking from memory, it won't be the same hash as the binary on disk
+        # since relocations have already be updated and linkage, yadayada
+             
+        vw.addFile(trimfname, va, fakemd5)
+        #FIXME: make sure all of everything is backed by a file...  vw.getFile() should always 
```

Comment:
Seems like an easy fix we can just add in?

Reply:
that was more a reminder for me to make sure it was working.